Lesson 6

ilya-burinskiy opened the pull request.

Commit:
Question resources

Commit:
Question controller

VidgarVii commented on app/controllers/questions_controller.rb:
```diff
@@ -0,0 +1,44 @@
+  end
+
+  def find_test
+    @test = Test.find_by!(id: params[:test_id])
```
Test.find(params[:test_id])

VidgarVii commented on app/controllers/questions_controller.rb:
```diff
@@ -0,0 +1,44 @@
+  end
+
+  def destroy
+    Question.find_by!(id: params[:id]).delete
```
должен быть @question.destroy!

Question.find_by! можно отдраить в метод.
delete не вызовет колбек на удаление ответов данного вопроса. По этому используют destroy

VidgarVii commented on app/controllers/questions_controller.rb:
```diff
@@ -0,0 +1,44 @@
+
+  def create
+    question = Question.create!(question_params)
+    render json: question.inspect
```
а если вопрос не создастся то выбросится ошибка. Что не хорошо. Нужна проверка.
```
question = Question.new(question_params)
if question.save
  redirect_to question
else
  render :new
end
```

VidgarVii commented on app/controllers/questions_controller.rb:
```diff
@@ -0,0 +1,44 @@
+  end
+
+  def show
+    render json: Question.find_by!(id: params[:id])
```
Question.find() 
и надо положить в метод, чтобы от сюда дергать только инстанс переменную

Commit:
Question controller fix

- Added find_question callback, for show and destroy methods
- Question is created only if it is valid

VidgarVii commented on app/controllers/questions_controller.rb:
```diff
@@ -0,0 +1,49 @@
+
+  def create
+    question = Question.new(question_params)
+    if question.valid?
```
различные логические выражения разделяй пустыми строками

Commit:
Final fixes
